command/agent/scada.go

```diff
@@ -0,0 +1,191 @@
+package agent
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/scada-client"
+)
+
+const (
+	// providerService is the service name we use
+	providerService = "consul"
+
+	// resourceType is the type of resource we represent
+	// when connecting to SCADA
+	resourceType = "infrastructures"
+)
+
+// ProviderService returns the service information for the provider
+func ProviderService(c *Config) *client.ProviderService {
+	return &client.ProviderService{
+		Service:        providerService,
+		ServiceVersion: fmt.Sprintf("%s%s", c.Version, c.VersionPrerelease),
+		Capabilities: map[string]int{
+			"http": 1,
+		},
+		Meta: map[string]string{
+			"auto-join":  strconv.FormatBool(c.AtlasJoin),
+			"datacenter": c.Datacenter,
+			"server":     strconv.FormatBool(c.Server),
+		},
+		ResourceType: resourceType,
+	}
+}
+
+// ProviderConfig returns the configuration for the SCADA provider
+func ProviderConfig(c *Config) *client.ProviderConfig {
+	return &client.ProviderConfig{
+		Service: ProviderService(c),
+		Handlers: map[string]client.CapabilityProvider{
+			"http": nil,
+		},
+		ResourceGroup: c.AtlasInfrastructure,
+		Token:         c.AtlasToken,
+	}
+}
+
+// NewProvider creates a new SCADA provider using the
+// given configuration. Requests are routed to the
```

Comment:
Sorry my comments are only related to formatting and comments :-), but this one jumped out at me as I was working my way through.  Comment just trails off.